src/codeu/chat/client/ClientConversation.java

```diff
@@ -51,21 +51,43 @@ public ClientConversation(Controller controller, View view, ClientUser userConte
     this.userContext = userContext;
   }
 
+  /** Rename the title of the conversation */
+  public void renameConversation(String newTitle) {
+    if (isValidTitle(newTitle)) {
+      controller.renameConversation(currentSummary.id, newTitle);
+    } else {
+      LOG.error("Error: new name is invalid.");
+    }
+  }
+
+  public void removeConversation(ConversationSummary cs) {
+    if (currentSummary != null) {
+      controller.removeConversation(cs.id);
+      summariesByUuid.remove(cs.id);
+      summariesSortedByTitle.remove(cs.getTitle());
+    }  
+  }
+
   public void setMessageContext(ClientMessage messageContext) {
     this.messageContext = messageContext;
   }
 
   // Validate the title of the conversation
-  static public boolean isValidTitle(String title) {
-    boolean clean = true;
+  public boolean isValidTitle(String title) {
     if ((title.length() <= 0) || (title.length() > 64)) {
-      clean = false;
-    } else {
+      return false;
+    } else if (titleExists(title)) {
```

Comment:
combine with the previous one?
if (title.length() <=0 || ... || titleExists(title)